clang/test/dpct/curand-cross-function-placeholder.cu

```diff
@@ -5,40 +5,29 @@
 #include <stdio.h>
 #include <curand.h>
 
-//CHECK: /*
-//CHECK-NEXT: DPCT1050:{{[0-9]+}}: The template argument of the RNG engine could not be deduced. You need to update this code.
-//CHECK-NEXT: */
-//CHECK-NEXT: void update(float* randvals, std::shared_ptr<dpct_placeholder/*Fix the engine type manually*/> rng, long long nx, long long ny) {
-//CHECK-NEXT:   oneapi::mkl::rng::uniform<float> distr_ct{{[0-9]+}};
-//CHECK-NEXT:   oneapi::mkl::rng::generate(distr_ct{{[0-9]+}}, *rng, nx*ny/2, randvals);
+//CHECK: void update(float* randvals, dpct::rng::host::rng_generator rng, long long nx, long long ny) {
+//CHECK-NEXT:   rng.generate_uniform(randvals, nx*ny/2);
 //CHECK-NEXT: }
 void update(float* randvals, curandGenerator_t rng, long long nx, long long ny) {
   curandGenerateUniform(rng, randvals, nx*ny/2);
 }
 
 
 //CHECK: int main(){
-//CHECK-NEXT:   dpct::device_ext &dev_ct1 = dpct::get_current_device();
-//CHECK-NEXT:   sycl::queue &q_ct1 = dev_ct1.default_queue();
-//CHECK-NEXT:   oneapi::mkl::rng::uniform<float> distr_ct{{[0-9]+}};
 //CHECK-NEXT:   long long nx = 5120;
 //CHECK-NEXT:   long long ny = 5120;
 //CHECK-NEXT:   unsigned long long seed = 1234ULL;
-//CHECK-NEXT:   std::shared_ptr<oneapi::mkl::rng::philox4x32x10> rng;
-//CHECK-NEXT:   std::shared_ptr<oneapi::mkl::rng::mrg32k3a> rng1;
-//CHECK-NEXT:   rng = std::make_shared<oneapi::mkl::rng::philox4x32x10>(q_ct1, seed);
-//CHECK-NEXT:   rng1 = std::make_shared<oneapi::mkl::rng::mrg32k3a>(q_ct1, seed);
-//CHECK-NEXT:   /*
-//CHECK-NEXT:   DPCT1026:{{[0-9]+}}: The call to curandSetPseudoRandomGeneratorSeed was removed because this call is redundant in SYCL.
-//CHECK-NEXT:   */
-//CHECK-NEXT:   /*
-//CHECK-NEXT:   DPCT1026:{{[0-9]+}}: The call to curandSetPseudoRandomGeneratorSeed was removed because this call is redundant in SYCL.
-//CHECK-NEXT:   */
+//CHECK-NEXT:   dpct::rng::host::rng_generator rng;
+//CHECK-NEXT:   dpct::rng::host::rng_generator rng1;
+//CHECK-NEXT:   (&rng)->create(dpct::rng::host::random_engine_type::philox4x32x10);
```

Comment:
Can we simplify all cases like that as follows?
```suggestion
//CHECK-NEXT:   rng.create(dpct::rng::host::random_engine_type::philox4x32x10);
```

Reply:
simplified